text/0000-raw-keywords.md

```diff
@@ -0,0 +1,270 @@
+- Feature Name: raw_keywords
+- Start Date: 2021-03-05
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+<!--
+One paragraph explanation of the feature.
+-->
+
+Reserve `k#keyword` in edition 2021 and beyond as a general syntax for adding keywords mid-edition instead of needing speculative reservations.
+
+# Motivation
+[motivation]: #motivation
+
+<!--
+Why are we doing this? What use cases does it support? What is the expected outcome?
+-->
+
+There were a few attempts to reserve keywords for the the 2018 edition.  Some of those proved controversial, and the language team eventually [decided](https://github.com/rust-lang/rfcs/pull/2441#issuecomment-395256368) not to accept any reservations for not-yet-approved features:
+
+> [...] felt particularly strongly that up-front reservations are wrong and a mistake in the initial Edition proposal, basically for the reasons I've already outlined in the thread: they force up-front decisions about surface issues of features that are not yet fully proposed, let alone accepted or implemented. That just seems totally backwards and is going to keep leading to unworkable discussions. We both feel that the role of Editions here is that they can absorb any keyword-flags that have accumulated in the meantime.
+>
+> In all, there is certainly no consensus to merge this RFC as-is, and I think there are no objections to instead closing it, under the assumption that we'll add a keyword-flag mechanism (or something like it) as needed later.
+
+This RFC is thus a proposal to add that general mechanism.
+
+The other thing that was learned with the 2018 edition is that the period between editions is long enough that the normal "stability without stagnation" principle of "it can just wait for the next train" doesn't work.  Instead, it encouraged rushing to try to get things in on time, which had negative quality of life consequences for many contributors.  As such, it's important that an alternative mechanism be made available so that missing an edition train doesn't mean having to wait another 3 years -- even if that alternative has syntax that's slightly less nice until the next train.
```

Comment:
I feel like the word "slightly" here is a subjective assertion that does not match my view. Something like `k#catch { ... }` looks quite poor to me, and I think will make documentation and teaching more difficult.  The `k#` prefix doesn't just go away during the next edition, it is around forever.  I can imagine trying to explain these things to novice users could increase the load when trying to learn a language that is already quite syntax-heavy.  It is also a small burden on tooling.

This also pre-supposes a 3-year cadence for editions, which is not mandated in #3085.  My impression is that features that add new keywords usually take years to develop (in the design, RFC, implementation, testing, stabilization, documentation steps). I would think that if something of that magnitude is ready in say 2023, a new edition release could be coordinated to ship it that year to avoid making the feature wait until 2024.

Reply:
> The `k#` prefix doesn't just go away during the next edition, it is around forever.

I'm not sure I follow that; for any given keyword, `k#` does go away during the next edition.